Fix crash in WebPageProxy::wheelEventHandlingCompleted()

smfr opened the pull request:
#### 530a64fb16571884b79d4e6c7eabbe3defdd4f4d
<pre>
Fix crash in WebPageProxy::wheelEventHandlingCompleted()
<a href="https://bugs.webkit.org/show_bug.cgi?id=258705">https://bugs.webkit.org/show_bug.cgi?id=258705</a>
rdar://111445101

Reviewed by Cameron McCormack.

Crash data show crashes in WebPageProxy::wheelEventHandlingCompleted(), and many of these are under WebProcessProxy::shutDown(),
suggesting that we&apos;ve closed the WebPageProxy. This is indeed possible, since WebPageProxy::sendWheelEvent() sends IPC
with an async reply.

So early return in the reply handler if closed, and also null-check the m_pageClient (which is a WeakPtr).

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::sendWheelEvent):
(WebKit::WebPageProxy::wheelEventHandlingCompleted):

Canonical link: <a href="https://commits.webkit.org/265817@main">https://commits.webkit.org/265817@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a79d0214cb1b2c68e317e5061e17fc6fcf945725

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11980 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12326 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12629 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13726 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11563 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14736 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12340 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14261 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12144 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12954 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10135 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14138 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10246 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17999 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11324 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11030 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14200 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11515 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/9475 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10729 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2930 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15054 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11373 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

heycam reviewed (approved):
There are other async calls in this file that do something with the page client in the callback, like in `WebPageProxy::beginTextRecognitionForVideoInElementFullScreen`. Do we need to handle them similarly?

smfr committed:
Fix crash in WebPageProxy::wheelEventHandlingCompleted()
https://bugs.webkit.org/show_bug.cgi?id=258705
rdar://111445101

Reviewed by Cameron McCormack.

Crash data show crashes in WebPageProxy::wheelEventHandlingCompleted(), and many of these are under WebProcessProxy::shutDown(),
suggesting that we've closed the WebPageProxy. This is indeed possible, since WebPageProxy::sendWheelEvent() sends IPC
with an async reply.

So early return in the reply handler if closed, and also null-check the m_pageClient (which is a WeakPtr).

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::sendWheelEvent):
(WebKit::WebPageProxy::wheelEventHandlingCompleted):

Canonical link: https://commits.webkit.org/265817@main
